index.html

```diff
@@ -977,13 +1003,50 @@
 				html = html + "<span class='varname'>Institute: </span> <span class='attribute'>" + data.institute + "</span>";
 				html = html + "<br />"
 				html = html + "<span class='varname'>Campus: </span> <span class='attribute'>" + standardizeCase(data.campus) + "</span>";
-				html = html + "<br />"
-				html = html + "<span class='varname'>School District: </span> <span class='attribute'>" + standardizeCase(data.district) + "</span>";
+				if (data.district) {
+					html = html + "<br />"
+					html = html + "<span class='varname'>School District: </span> <span class='attribute'>" + standardizeCase(data.district) + "</span>";
+				}
 				return html;
 				//this will return the string to the calling function
 
 			}
 
+// These are the popups for the polygon district layers, using Eldan's House/Senate 'show' logic
+// When a click event occurs on a feature in the unioned districts layer, open a popup for
+// the correct district type at the location of the click, with description HTML from its properties.
+			function fillpopup(data) {
+				var html = "<span class='varname'>";
+// the shorthand in this next line is just a compressed if...then.
+// "if showHouseDistricts is true then use the first string, else the second"
+				html += showHouseDistricts ? "House District: " : "Senate District: ";
+				html += "</span><span class='attribute'>";
+				html += showHouseDistricts ? data.HseDistNum : data.SenDistNum;
+				html += "</span>";
+				return html; //this will return the string to the calling function
+			}
+
+			map.on('click', 'school_house_senate_districts_UNION-poly', function (e) {
+				var pointFeatures = map.queryRenderedFeatures(e.point, {layers: loadedPointLayerNames});
```

Comment:
This is the important bit.  `queryRenderedFeatures` tells us every layer that the click has hit.  Adding the `{layers: loadedPointLayerNames}` filter lets us get only our point layers, not mapbox built-in ones like "town name".  Then the conditional in the next line stops this popup from being drawn if any of our point layers were returned.